auth0/src/main/java/com/auth0/android/authentication/storage/CryptoUtil.java

```diff
@@ -145,6 +146,22 @@ public CryptoUtil(@NonNull Context context, @NonNull Storage storage, @NonNull S
         }
     }
 
+    private KeyStore.PrivateKeyEntry getKeyEntryCompat(KeyStore keyStore)
+            throws KeyStoreException, NoSuchAlgorithmException, UnrecoverableEntryException {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            //Following code is for API 28+
+            PrivateKey privateKey = (PrivateKey) keyStore.getKey(KEY_ALIAS, null);
+            Certificate certificate = keyStore.getCertificate(KEY_ALIAS);
```

Comment:
this is only used for the `new KeyStore.PrivateKeyEntry(privateKey, new Certificate[]{certificate});` call, we could avoid calling this if the if-case doesn't require it.

Reply:
@lbalmaceda Are you referring to the `Certificate` object? Since the `privateKey` check was moved before it, now it is only created when needed. Or did you mean I should create it inline as `new Certificate[] {keystore.getCertificate(KEY_ALIAS)}`?